Sources/NIO/BaseSocket.swift

```diff
@@ -301,6 +301,29 @@ class BaseSocket: BaseSocketProtocol {
         }
         return sock
     }
+    
+    /// Cleanup the unix domain socket.
+    ///
+    /// Deletes the associated file if it exists and has socket type.
+    ///
+    /// - parameters:
+    ///     - unixDomainSocketPath: The pathname of the UDS.
+    /// - throws: An `IOError` if the operation failed.
+    static func cleanupSocket(unixDomainSocketPath: String) throws {
+        let sb: UnsafeMutablePointer<stat> = UnsafeMutablePointer<stat>.allocate(capacity: 1)
+        if lstat(unixDomainSocketPath, sb) == 0 {
```

Comment:
We have wrapped all our syscalls in helpers to deal with things like `EINTR`. This is necessary here as well. See `System.swift` which has a definition of `stat`. If you think using `lstat` is necessary here, then we need an equivalent wrapper.

Reply:
introduced `lstat` wrapper - current `fstat` expects file descriptor that we don't have in this scenario